cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -353,6 +378,46 @@ func (m *azureCache) getAutoscalingOptions(ref azureRef) map[string]string {
 	return m.autoscalingOptions[ref]
 }
 
+// HasInstance returns if a given instance exists in the azure cache
+func (m *azureCache) HasInstance(providerID string) (bool, error) {
+	vmsPoolSet := m.getVMsPoolSet()
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	resourceID, err := convertResourceGroupNameToLower(providerID)
```

Comment:
this covertResourceGroupNameToLower, alongside the validation inside could be split out into our ProviderIDParser and a 
Validate() call could happen at the top of hasinstance, and here we just covert to lowercase to normalize it.